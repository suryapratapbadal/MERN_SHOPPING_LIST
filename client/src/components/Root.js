import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css'
import AppNavbar from './AppNavbar';
import ShoppingList from './ShoppingList';
import ItemModal from './ItemModal';
import LogIn from './LogIn';
import { Container } from 'reactstrap';
import { connect } from 'react-redux';


class Root extends Component {
    constructor(props) {
        super(props);
        this.state = {
            user: this.props.user,
        };
    }
    render() {  
        return (
            <div className="App">
                <AppNavbar changeTheme={this.props.changeTheme}/>
                {
                    !this.props.user ? <LogIn /> :
                    <Container>
                        <ItemModal/>
                        <ShoppingList/>
                    </Container>
                }
            </div>
        );
    }
}

export default connect(state => {
    const user = state.itemReducer.user || false;
    const loading = state.itemReducer.loading;
    return {
        user,
        loading
    }
})(Root);
