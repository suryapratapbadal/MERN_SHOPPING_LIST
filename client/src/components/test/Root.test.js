import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import { Root } from '../Root';



// Component Renders

describe('<Root/>', () => {

    const wrapper = shallow(<Root user={false} loading={false}/>);

    describe('render()', () => {

        test('renders login component', () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });


    });
});