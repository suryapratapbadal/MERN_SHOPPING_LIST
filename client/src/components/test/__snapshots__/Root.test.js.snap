// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root/> render() renders login component 1`] = `
<div
  className="App"
>
  <Connect(WithStyles(AppNavbar)) />
  <Connect(WithStyles(LogIn)) />
</div>
`;
